stub_uploader/get_version.py

```diff
@@ -62,13 +66,26 @@ def compute_incremented_version(
     # version will satisfy the version_spec (defined precisely by the `compatible`
     # specifier below). This allows users to have expectations of what a stub package
     # will contain based on the upstream version they're targeting.
-    if version_spec.endswith(".*"):
+    m = POST_SPEC_RE.match(version_spec)
+    if m:
+        compatible = SpecifierSet(f"=={m.group(1)}.{m.group(2)}.*")
```

Comment:
Similarly, this could use `Version(version_spec.removesuffix(".*")).base_version` here